### PR TITLE
Keep backward compatibility on cupy.cudnn.batch_normalization_forward_training

### PR DESCRIPTION
**Pull request by takagi**
This PR follows #2060 to make it keep backward compatibility.

### CODE REVIEW

**Comment by kmaehashi** [comment]
Is this PR still WIP?
Currently Chainer tests are failing, so I think it's better to merge this ASAP.

**Comment by takagi** [comment]
I removed WIP.

File: cupy/cudnn.pyx [hunk 1]
```diff
@@ -1570,8 +1570,14 @@ cdef _get_dtype_of_tensor_descriptor(size_t desc):
 def batch_normalization_forward_training(
         core.ndarray x, core.ndarray gamma, core.ndarray beta,
         core.ndarray running_mean, core.ndarray running_var,
-        double eps, double decay,
+        mean, inv_std, double eps, double decay,
         bint is_for_conv2d, int cudnn_mode, bint debug):
+    # Usually supply None to mean and inv_std, which are left for backward
+    # compatibility. See cupy#2060 and cupy#2070.
+    if (mean is None and inv_std is not None or
+            inv_std is None and mean is not None):
```
Comment (toslunar) [inline cupy/cudnn.pyx]: How about `if (mean is None) != (inv_std is None):`?

Comment (takagi) [reply cupy/cudnn.pyx]: Smart, fixed!

**Comment by toslunar** [comment]
Jenkins, test this please.

**Comment by toslunar** [comment]
Chainer needs to be fixed simultaneously.

**Pull request merged**